refactor: move dwt_denoise into common/dwt.c

ralfbrown opened the pull request:
...where it belongs and would have been originally if not for another pending PR on dwt.c.
Also put dwt_interleave_rows into dwt.h where it will be available to filmicrgb.c for an upcoming PR.

Finally, refactor the X-Trans codepath in rawdenoise.c to use the new common function for computing noise thresholds.

TurboGit commented on src/iop/rawdenoise.c:
```diff
@@ -481,48 +320,16 @@ static void dwt_denoise(float *const img, const int width, const int height,
   const int height = roi->height;
   const size_t size = (size_t)width * height;
-  float *const fimg = malloc((size_t)size * 4 * sizeof(float));
+  float *const fimg = dt_alloc_sse_ps((size_t)size * 4);
```
dt_alloc_sse_ps -> dt_alloc_align_float.

TurboGit commented on src/common/dwt.c:
```diff
@@ -426,6 +409,151 @@ void dwt_decompose(dwt_params_t *p, _dwt_layer_func layer_func)
+ */
+void dwt_denoise(float *const img, const int width, const int height, const int bands, const float *const noise)
+{
+  float *const details = dt_alloc_sse_ps(2 * width * height);
```
dt_alloc_align_float

TurboGit reviewed (changes requested):
Minor clean-up for float alloc. TIA.

ralfbrown commented:
Done.

TurboGit commented:
@ralfbrown  : rawdenoise conflcts due to merging your other PR.

ralfbrown commented:
Rebased, and I've also switched over the X-Trans code path to the new common code.  X-Trans now shows 5.3x speedup single-threaded, 1.8x at 8 threads, but only 1.6 percent speedup at 32 threads because the memory bus is being saturated.  Output verified against the updated test 0068.

TurboGit reviewed (approved):
Thanks ! All good on my side.
